core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -602,18 +603,62 @@ private static AnnotatedValueResolver ofQueryParamMap(String name,
                                                           AnnotatedElement annotatedElement,
                                                           AnnotatedElement typeElement, Class<?> type,
                                                           DescriptionInfo description) {
+        final Type valueType = ((ParameterizedType) ((Parameter) typeElement).getParameterizedType())
+                .getActualTypeArguments()[1];
+        final Class<?> rawValueType = ClassUtil.typeToClass(valueType);
+        assert rawValueType != null;
+
+        if (valueType instanceof ParameterizedType && !(rawValueType == Iterable.class ||
+                                                        rawValueType == List.class ||
+                                                        rawValueType == Collection.class ||
+                                                        rawValueType == Set.class)) {
+            throw new IllegalArgumentException(
+                    "Invalid Map value type: " + rawValueType +
+                    " (expected Iterable, List, Collection or Set)");
+        }
```

Comment:
Question) Would it make more sense to just support list or set since that's all we actually support? e.g. Users specifying `Map<String, Deque<Object>>` will run into a casting exception later on anyways

```suggestion
        if (valueType instanceof ParameterizedType && !(List.class.isAssignableFrom(rawValueType) || Set.class.isAssignableFrom(rawValueType))) {
            throw new IllegalArgumentException(
                    "Invalid parameterized map value type: " + rawValueType +
                    " (expected List or Set)");
        }
```